pal/src/misc/bstr.cpp

```diff
@@ -130,16 +130,20 @@ STDAPI_(BSTR) SysAllocStringLen(const OLECHAR *psz, UINT len)
  *  return value = void
  *
  ***********************************************************************/
-STDAPI_(void) SysFreeString(const OLECHAR* psz)
+STDAPI_(void) SysFreeString(BSTR bstr)
 {
-    if (psz != NULL)
+    if (bstr != NULL)
     {
-        HeapFree(GetProcessHeap(), 0, (LPVOID) psz);
+        bstr = (BSTR) ((char*) bstr - sizeof(DWORD));
+#if defined(_WIN64)
+        bstr = (BSTR) ((char*) bstr - sizeof(DWORD));
```

Comment:
This is because of extra offset done for `_WIN64` on line 105. Actually I don't understand line 105 or its comment. Maybe it was done for alignment on x64? It looks without the extra offset it works fine on Linux Amd64.